### PR TITLE
Feature #354 xASL_io_SplitASL_M0: Remove asl dummy

### LINKED ISSUE #354: Option to skip dummy ASL scans
Create an option for skipping ASL scans.
These can be used to skip the dummy and noise scans Wibeke has.

### Release summary
Added an option x.DummyPositionInASL4D that removes marked dummy scans when splitting ASL to ASL+M0+dummy

### PR DESCRIPTION
**Pull request by jan-petr**
Closes #354

### CODE REVIEW

File: DataParTemplate.m
Comment (HenkMutsaerts) [inline DataParTemplate.m:90]: here the format of the parameter is missing (numerical, scalar or vector)

File: Functions/xASL_io_SplitASL_M0.m
Comment (HenkMutsaerts) [inline Functions/xASL_io_SplitASL_M0.m:1]: Would rename this function to `xASL_io_SplitASL`

File: Functions/xASL_io_SplitASL_M0.m
Comment (HenkMutsaerts) [inline Functions/xASL_io_SplitASL_M0.m:25]: This should be:
`%EXAMPLE for moving the first two volumes to M0.nii`

File: Functions/xASL_io_SplitASL_M0.m
Comment (HenkMutsaerts) [inline Functions/xASL_io_SplitASL_M0.m:29]: This is incorrect right, in this example, the M0 is not moved but only the dummy volume is removed?
Also, I would make it more specific by writing "removed" rather than "excluded".

File: Functions/xASL_io_SplitASL_M0.m
Comment (HenkMutsaerts) [inline Functions/xASL_io_SplitASL_M0.m:41]: you can probably omit the `if ischar` as `xASL_str2num` has this check built-in

File: Functions/xASL_io_SplitASL_M0.m
Comment (HenkMutsaerts) [inline Functions/xASL_io_SplitASL_M0.m:44]: This could allow a matrix instead of vector right?

File: Functions/xASL_io_SplitASL_M0.m
Comment (HenkMutsaerts) [reply Functions/xASL_io_SplitASL_M0.m:41]: Same below

File: Modules/xASL_module_ASL.m
Comment (HenkMutsaerts) [inline Modules/xASL_module_ASL.m:95]: Is this part close to settings the default for these parameters (should also be the beginning of `xASL_module_ASL`)? Just to avoid bulky code

**Review by HenkMutsaerts — APPROVED** [review]
Few minor things, looks good otherwise

File: DataParTemplate.m
Comment (jan-petr) [reply DataParTemplate.m:90]: Updated of M0Position. And added an example for null entry (as discussed with Maurice).

File: Functions/xASL_io_SplitASL_M0.m
Comment (jan-petr) [reply Functions/xASL_io_SplitASL_M0.m:1]: Agree. Done.

File: Functions/xASL_io_SplitASL_M0.m
Comment (jan-petr) [reply Functions/xASL_io_SplitASL_M0.m:25]: Done.

File: Functions/xASL_io_SplitASL_M0.m
Comment (jan-petr) [reply Functions/xASL_io_SplitASL_M0.m:29]: Good point. Updated examples.

File: Functions/xASL_io_SplitASL_M0.m
Comment (jan-petr) [reply Functions/xASL_io_SplitASL_M0.m:41]: Agree. Thanks.

File: Functions/xASL_io_SplitASL_M0.m
Comment (jan-petr) [reply Functions/xASL_io_SplitASL_M0.m:44]: yep a matrix will be converted to a vector. This should be fine, right?

File: Modules/xASL_module_ASL.m
Comment (jan-petr) [reply Modules/xASL_module_ASL.m:95]: Defaults are defined above. This was only checking if it makes sense to run the function or not. But this is now moved inside this wrapper. If no indices are provided, it will not continue running the SplitASL function.

**Comment by jan-petr** [comment]
All comments fixed. Check the two unresolved comments.

File: Functions/xASL_io_SplitASL_M0.m
Comment (HenkMutsaerts) [reply Functions/xASL_io_SplitASL_M0.m:46]: No, my point was that a matrix should give a warning or error. So it is fine to force the vector direction (horizontal vs vertical), but this way you explicitly allow a wrong input (i.e. a matrix).

E.g., if iMo = [1 2 3;1 2 3] it will become [1 2 3 1 2 3] without a warning or error

File: DataParTemplate.m
Comment (HenkMutsaerts) [reply DataParTemplate.m]: What does "Updated of M0Position" mean? ;)

The null entry is fine if it is JSON and ExploreASL `[]` standard, I would only ensure that advanced parameters are not in the example JSON file.

Comment (jan-petr) [reply DataParTemplate.m]: Haha - that's a fusion of "Update of M0Position" and "Updated M0Position" and I wanted to say that I have updated the description of the M0PositionInASL4D. It was late :) Comment sloppy, update itself not sloppy :)

Agree - DummyPositionInASL4D wasn't added to the example JSON.

File: Functions/xASL_io_SplitASL_M0.m
Comment (jan-petr) [reply Functions/xASL_io_SplitASL_M0.m:46]: Yes, good point. If size(iM0,1)>1 && size(iM0,2)>1 then a warning... 
I'll update that.

**Comment by jan-petr** [comment]
All should be fixed. Need your formal approval before merging.

**Review by HenkMutsaerts — APPROVED** [review]
Great, yes this was rather something I noted. We should do a formal input parameter check as Pieter suggested before. Let's put this on the list :)

**Comment by jan-petr** [comment]
> Great, yes this was rather something I noted. We should do a formal input parameter check as Pieter suggested before. Let's put this on the list :)

Yes, we should. But we do not suck so much in parameter checking anymore. So other things - unit testing, Flavors testing, new features have much higher priority :)

**Pull request merged**
